### PR TITLE
fix(server): recover direct stalled reviews in place

### PR DESCRIPTION
**Pull request by cryppadotta**
## Thinking Path

> - Paperclip is the open source control plane for AI-agent companies.
> - The recovery service keeps issue graphs live when agent work stops.
> - An issue in review must keep a clear action path until someone resolves it.
> - A direct stalled review can be both the source issue and the recovery target.
> - The generic liveness path can otherwise suggest a nested recovery issue for the same work.
> - This pull request blocks the direct review in place, records the recovery action, and wakes its selected owner with a normal model.
> - The benefit is bounded recovery without recursive child work or a silent stalled review.

## Linked Issues or Issue Description

**What happened?**

The liveness backstop treated a directly stalled `in_review` issue like a separate recovery target. The resulting action could suggest nested recovery work even though the source and target were the same issue.

**Expected behavior**

Paperclip must block the direct review in place. It must leave a clear system notice, wake the selected recovery owner, and avoid a dedicated child issue.

**Steps to reproduce**

1. Create an agent-owned issue with status `in_review`.
2. Leave the issue without a maintained review or decision path.
3. Run issue-graph liveness reconciliation.
4. Observe that the source issue needs in-place recovery and must not create a nested recovery child.

**Paperclip version or commit**

`master` at `536d5880c5` before this pull request.

**Deployment mode**

Core server behavior. The issue is not deployment-specific.

**Installation method**

Built from source with pnpm.

**Agent adapter(s) involved**

Not adapter-specific. The bug is in core recovery behavior.

**Database mode**

Both embedded PGlite and external PostgreSQL use this service path.

Refs #10675

## What Changed

- Detect the direct source-and-recovery-target case in liveness action guidance.
- Block a directly stalled review in place and add a structured recovery notice.
- Wake the selected recovery owner without creating a nested child issue.
- Keep the in-place recovery wake on the normal model because it can require review work.
- Add regression coverage for direct-review recovery, recovery fallbacks, and agent secret-binding updates.

## Verification

- `pnpm -r typecheck`
- `pnpm test:run`
- `pnpm build`
- Focused direct-review recovery regression: 1 passed, 23 skipped.
- Focused recovery and secret-binding regression set: 5 passed, 126 skipped.

## Risks

- Low risk. The behavior changes only the direct stalled-review edge case.
- The issue now moves to `blocked` and can generate one owner wake. The regression test verifies that no child recovery issue is created.
- There are no schema, migration, API-contract, or UI changes.

> For core feature work, check [`ROADMAP.md`](ROADMAP.md) first and discuss it in `#dev` before opening the PR. Feature PRs that overlap with planned core work may need to be redirected — check the roadmap first. See `CONTRIBUTING.md`.

## Model Used

OpenAI Codex, GPT-5 family. The runtime does not expose the exact deployment ID or context-window size. The model used reasoning, repository tools, and code execution.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have searched GitHub for duplicate or related PRs and linked them above
- [x] I have either (a) linked existing issues with `Fixes: #` / `Closes #` / `Refs #` OR (b) described the issue in-PR following the relevant issue template
- [x] I have not referenced internal/instance-local Paperclip issues or links (only public GitHub `#NNN` / `github.com/paperclipai/paperclip` URLs)
- [x] My branch name describes the change (e.g. `docs/...`, `fix/...`) and contains no internal Paperclip ticket id or instance-derived details
- [x] I have run tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] I have updated relevant documentation to reflect my changes
- [x] I have considered and documented any risks above
- [x] All Paperclip CI gates are green
- [x] Greptile is 5/5 with no open P2s, recommendations, or follow-ups
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

File: server/src/services/recovery/service.ts [hunk 1]
```diff
@@ -4991,6 +5012,52 @@ export function recoveryService(db: Db, deps: { enqueueWakeup: RecoveryWakeup })
...
+          sourceIssueId: updated.id,
+          recoveryIssueIdentifier: recoveryIssue.identifier,
+          recoveryIssueId: recoveryIssue.id,
+          finding: input.finding,
+          prefix,
+        }),
+        { runId: input.runId ?? null },
+      );
+
+      await deps.enqueueWakeup(ownerSelection.agentId, {
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply server/src/services/recovery/service.ts:5039]: Fixed in d091740714. Direct in-place recovery now reassigns the blocked review to the selected fallback owner before queuing that owner wake. The regression test forces the original assignee over budget and verifies that the manager is both the new assignee and the sole wake target.

**Comment by AlexGogin** [comment]
Production confirmation on managed npm `2026.831.1`: an executor moved an issue to `in_review` but created no interaction, approval, review participant, or valid external-wait monitor. The manager received no event and the card remained idle indefinitely while looking healthy on the board.

Our temporary five-minute recovery pass now detects `in_review` older than 45 minutes with no pending interaction/approval, no live run/wake, and no `waitingOnExternal=true` monitor, then routes it to the assignee's manager with an issue-scoped wake. It recovered the incident immediately.

This PR addresses the exact platform gap. Please retain a regression case for a bare `in_review` issue: `issue-graph-liveness.ts` already classifies `in_review_without_action_path`, but current reconciliation skips it when there is no pending execution participant.
